lib/streamlit/commands/execution_control.py

```diff
@@ -87,3 +90,51 @@ def experimental_rerun() -> NoReturn:
     # be seen.
     _LOGGER.warning(msg)
     rerun()
+
+
+@gather_metrics("switch_page")
+def switch_page(page: str) -> NoReturn:  # type: ignore[misc]
+    """Switch the current programmatically page in a multi-page app.
+    When `st.switch_page()` is called with a page, the current page script is halted
+    and the requested page script will be queued to run from the top.
+    Parameters
+    ----------
+    page: str
+        The file path, relative to the main script, of the page to switch to.
+    """
+
+    ctx = get_script_run_ctx()
+
+    ctx_main_script = ""
+    if ctx:
+        ctx_main_script = ctx.main_script_path
+
+    main_script_path = os.path.join(os.getcwd(), ctx_main_script)
+    main_script_directory = os.path.dirname(main_script_path)
+
+    # Convenience for handling ./ notation and ensure leading / doesn't refer to root directory
+    page = os.path.normpath(page.strip("/"))
+
+    # Build full path
+    requested_page = os.path.join(main_script_directory, page)
+    all_app_pages = source_util.get_pages(ctx_main_script).values()
+
+    page_found = False
+    for page_data in all_app_pages:
+        full_path = page_data["script_path"]
+
+        if requested_page == full_path and ctx and ctx.script_requests:
+            page_found = True
+            ctx.script_requests.request_rerun(
+                RerunData(
+                    query_string="",
+                    page_script_hash=page_data["page_script_hash"],
+                )
+            )
+            # Force a yield point so the runner can do the rerun
+            st.empty()
```

Comment:
Why not just return here and avoid the `page_found` logic? Assuming the loop exists and a return hasn't been called.